contracts/src/main/kotlin/com/r3/developers/apples/contracts/AppleStampContract.kt

```diff
@@ -0,0 +1,31 @@
+package com.r3.developers.apples.contracts
+
+import com.r3.developers.apples.states.AppleStamp
+import net.corda.v5.ledger.utxo.Contract
+import net.corda.v5.ledger.utxo.transaction.UtxoLedgerTransaction
+
+class AppleStampContract: Contract {
+    override fun verify(transaction: UtxoLedgerTransaction) {
+        // Extract the command from the transaction
+        // Verify the transaction according to the intention of the transaction
+        when (val command = transaction.commands.first()) {
+            is AppleCommands.Issue -> {
+                val output = transaction.getOutputStates(AppleStamp::class.java).first()
+                require(transaction.outputContractStates.size == 1) {
+                    "This transaction should only have one AppleStamp state as output"
+                }
+                require(output.stampDesc.isNotBlank()) {
+                    "The output AppleStamp state should have clear description of the type of redeemable goods"
+                }
+            }
+            is AppleCommands.Redeem -> {
+                // Transaction verification will happen in BasketOfApplesContract
```

Comment:
This is interesting.
Is there a way I could redeem an apples stamp with an invalid transaction?
Right now we can only redeem a stamp for a basket of apples, sure; but what if in future Apple Farm decides to sell apples by the bag, and individually, as well as by the basket? Just hypothetical. Maybe this will be answered by Matt

Reply:
documented in https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4538564702/New+Joiner+DevEx+Developers+-+Feedback so that this thought doesn't get lost. resolving conversation